hazelcast/include/hazelcast/client/serialization/pimpl/compact.i.h

```diff
@@ -220,6 +220,88 @@ compact_reader::read()
     return values;
 }
 
+template<typename T>
+typename std::enable_if<
+  std::is_same<big_decimal, typename std::remove_cv<T>::type>::value,
+  typename boost::optional<T>>::type
+compact_reader::read()
+{
+    int32_t size = object_data_input.read<int32_t>();
+    std::vector<int8_t> bytes(size);
+    object_data_input.read_fully(bytes);
+    auto cppInt = client::pimpl::from_bytes(std::move(bytes));
```

Comment:
Define and use `read<big_integer>` to match the Java code. Similarly, for the write part.

Reply:
Again we will have the problem of returning optional from read.